dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java

```diff
@@ -5336,4 +5341,757 @@ public void JwtBuilderAPIBasicTests_multiple_apis_and_consumeToken() throws Exce
+        Expectations expectations = BuilderHelpers.createGoodBuilderExpectations(JWTBuilderConstants.JWT_BUILDER_SETAPIS_ENDPOINT, baseSettings, builderServer);
+
+        // build settings that will tell the test app how to run/what to pass to the "claimFrom" api
+        // give it a flag that says jwt string, and then pass the 3 part jwt token string
+        JSONObject testSettings = new JSONObject();
+        testSettings.put(JWTBuilderConstants.JWT_BUILDER_ACTION_CLAIM_FROM, JWTBuilderConstants.JWT_BUILDER_ACTION_CLAIM_FROM_JWT_STRING);
+        testSettings.put(JWTBuilderConstants.JWT_TOKEN, jwtToken);
+
+        // Now, add any override values - for this test, there are none
+        // Invoke the builder app to create a token (using (a second) builder, load all claims from the token into the second builder
+        String builderId = "encryptJwtRS256";
+        Page response = actions.invokeJwtBuilder_setApis(_testName, builderServer, builderId, testSettings);
+
+        validationUtils.validateResult(response, expectations);
+
+    }
+
+    // TODO Failing since claimsFrom doesn't know how to parse JWE Token
+    //@Test
+    public void JwtBuilderAPIBasicTests_claimsFrom_jweString_to_jwtToken_allClaims() throws Exception {
+
+        // build a token using the alternate builder config (we'll get claims from it)
+        String baseBuilderId = "encryptJwtRS256";
+        String jwtToken = getBaseToken(baseBuilderId);
+        JSONObject baseSettings = BuilderHelpers.setClaimsFromToken(jwtToken, JwtKeyTools.getComplexPrivateKeyForSigAlg(builderServer, JWTBuilderConstants.SIGALG_RS256));
+        // since we getting all claims in the case, we can base our expectations upon the content of this first token)
+        Expectations expectations = BuilderHelpers.createGoodBuilderExpectations(JWTBuilderConstants.JWT_BUILDER_SETAPIS_ENDPOINT, baseSettings, builderServer);
+
+        // build settings that will tell the test app how to run/what to pass to the "claimFrom" api
+        // give it a flag that says jwt string, and then pass the 3 part jwt token string
+        JSONObject testSettings = new JSONObject();
+        testSettings.put(JWTBuilderConstants.JWT_BUILDER_ACTION_CLAIM_FROM, JWTBuilderConstants.JWT_BUILDER_ACTION_CLAIM_FROM_JWT_STRING);
+        testSettings.put(JWTBuilderConstants.JWT_TOKEN, jwtToken);
+
+        // Now, add any override values - for this test, there are none
+        // Invoke the builder app to create a token (using (a second) builder, load all claims from the token into the second builder
+        String builderId = "jwt1";
+        Page response = actions.invokeJwtBuilder_setApis(_testName, builderServer, builderId, testSettings);
+
+        validationUtils.validateResult(response, expectations);
+
+    }
+
+    // TODO Failing since claimsFrom doesn't know how to parse JWE Token
+    //@Test
+    public void JwtBuilderAPIBasicTests_claimsFrom_jweStringRS256_to_jweTokenRS384_allClaims() throws Exception {
```

Comment:
Shouldn't need this test since `JwtBuilderAPIBasicTests_claimsFrom_jweString_to_jwtToken_allClaims` should be doing the same thing. We may need to consider a TODO on the product code side to decide whether we want to detect a possible JWE being passed in and issue an error message, in addition to updating the Javadoc.

Reply:
clean up of this will come with the next delivery...